### PR TITLE
Feat/shell routing plugin

### PR DESCRIPTION
**Pull request by frenautvh**
| Question         | Answer
| ---------------- | ---
| Branch?          | feat/manager-shell
| Bug fix?         | no
| New feature?     | yes
| Breaking change? | no
| Tickets          | MANAGER-7750
| License          | BSD 3-Clause

move shell routing into a plugin

### CODE REVIEW

File: packages/components/ovh-shell/package.json [hunk 1]
```diff
@@ -11,17 +11,20 @@
...
-  "exports": {
-    ".": "./dist/index.js",
-    "./client": "./dist/client/index.js",
-    "./plugin/i18n": "./dist/plugin/i18n/index.js"
-  },
   "main": "./dist/index.js",
   "scripts": {
-    "build": "tsc"
+    "build": "tsc",
+    "dev": "tsc -w"
```
Comment (mohammed-zahaf) [inline packages/components/ovh-shell/package.json:17]: i try to found `-w` option into the typescript compiler documentation but nothing.
is't sure this option is available ?

I found this `tsc --watch`

Comment (antleblanc) [reply packages/components/ovh-shell/package.json:17]: It's a shorthand

```sh
$ npx tsc --help
Version 3.8.3
Syntax:   tsc [options] [file...]

Examples: tsc hello.ts
          tsc --outFile file.js file.ts
          tsc @args.txt
          tsc --build tsconfig.json

Options:
 -h, --help                                         Print this message.
 -w, --watch                                        Watch input files.
```

;)

File: packages/components/ovh-shell/src/client/index.ts [hunk 2]
```diff
@@ -1,8 +1,10 @@
 import ShellClient from './shell-client';
 import exposeApi from './api';
 
-export default function useShellApi() {
+export function useShellApi() {
```
Comment (Jisay) [inline packages/components/ovh-shell/src/client/index.ts:4]: Any reason of not using arrow functions?

**Review by Jisay — APPROVED** [review]

File: packages/components/ovh-shell/src/client/index.ts (hunk 2 above)
Comment (zakariasahmane) [reply packages/components/ovh-shell/src/client/index.ts:4]: I don't think it changes anything here, the function relies on its own context so using arrow functions doesn't have any added value

**Review by zakariasahmane — APPROVED** [review]

File: packages/components/ovh-shell/src/client/index.ts (hunk 2 above)
Comment (frenautvh) [reply packages/components/ovh-shell/src/client/index.ts:4]: yes i think it's a matter of preferences, i believe that using named regular function might facilitates debugging as you get the function name in the debugger but that's pretty much it

**Pull request merged**